### ISSUE TITLE
Build & Runtime Errors

### ISSUE THREAD

**Issue by evoactivity** [jelhan/webpack-subresource-integrity-embroider#4]
I tried to give this a go but immediately ran into build errors. Seems to have trouble finding these files. I tried twice and it was a different file both times.

```
Build Error (PackagerRunner)

ENOENT: no such file or directory, open '/var/folders/rw/zh799lvx7lzb46z1mh_r851m0000gn/T/broccoli-27890EVXZIM2sCKQ2/out-1227-packager_runner_embroider_webpack/admin/ember-cli-live-reload.js'


Stack Trace and Error Report: /var/folders/rw/zh799lvx7lzb46z1mh_r851m0000gn/T/error.dump.8bb1a940ab738b0008b6a5d7bb0d9f89.log
```

```
Build Error (PackagerRunner)

ENOENT: no such file or directory, open '/var/folders/rw/zh799lvx7lzb46z1mh_r851m0000gn/T/broccoli-291856FKT9luhn2yN/out-1227-packager_runner_embroider_webpack/admin/assets/vendor.js'


Stack Trace and Error Report: /var/folders/rw/zh799lvx7lzb46z1mh_r851m0000gn/T/error.dump.116f0f8d609fa23b64014e03a8e8d80f.log
```

<details>
  <summary>Error Dump</summary>

  ```
=================================================================================

ENV Summary:

  TIME: Sat Feb 17 2024 14:04:59 GMT+0000 (Greenwich Mean Time)
  TITLE: ember
  ARGV:
  - /Users/liam/.volta/tools/image/node/18.17.0/bin/node
  - /Users/liam/Work/project/node_modules/.pnpm/ember-cli@5.5.0/node_modules/ember-cli/bin/ember
  - serve
  EXEC_PATH: /Users/liam/.volta/tools/image/node/18.17.0/bin/node
  TMPDIR: /var/folders/rw/zh799lvx7lzb46z1mh_r851m0000gn/T
  SHELL: /opt/homebrew/bin/fish
  PATH:
  - stuff
  PLATFORM: darwin arm64
  FREEMEM: 176603136
  TOTALMEM: 68719476736
  UPTIME: 3031109
  LOADAVG: 6.802734375,5.9326171875,5.4384765625
  CPUS:
  - Apple M1 Max - 24
  - Apple M1 Max - 24
  - Apple M1 Max - 24
  - Apple M1 Max - 24
  - Apple M1 Max - 24
  - Apple M1 Max - 24
  - Apple M1 Max - 24
  - Apple M1 Max - 24
  - Apple M1 Max - 24
  - Apple M1 Max - 24
  ENDIANNESS: LE
  VERSIONS:
  - acorn: 8.8.2
  - ada: 2.5.0
  - ares: 1.19.1
  - brotli: 1.0.9
  - cldr: 43.0
  - icu: 73.1
  - llhttp: 6.0.11
  - modules: 108
  - napi: 9
  - nghttp2: 1.52.0
  - nghttp3: 0.7.0
  - ngtcp2: 0.8.1
  - node: 18.17.0
  - openssl: 3.0.9+quic
  - simdutf: 3.2.12
  - tz: 2023c
  - undici: 5.22.1
  - unicode: 15.0
  - uv: 1.44.2
  - uvwasi: 0.0.18
  - v8: 10.2.154.26-node.26
  - zlib: 1.2.13.1-motley

ERROR Summary:

  - broccoliBuilderErrorStack: Error: ENOENT: no such file or directory, open '/var/folders/rw/zh799lvx7lzb46z1mh_r851m0000gn/T/broccoli-27890EVXZIM2sCKQ2/out-1227-packager_runner_embroider_webpack/admin/ember-cli-live-reload.js'
  - code: [undefined]
  - codeFrame: ENOENT: no such file or directory, open '/var/folders/rw/zh799lvx7lzb46z1mh_r851m0000gn/T/broccoli-27890EVXZIM2sCKQ2/out-1227-packager_runner_embroider_webpack/admin/ember-cli-live-reload.js'
  - errorMessage: ENOENT: no such file or directory, open '/var/folders/rw/zh799lvx7lzb46z1mh_r851m0000gn/T/broccoli-27890EVXZIM2sCKQ2/out-1227-packager_runner_embroider_webpack/admin/ember-cli-live-reload.js'
        at PackagerRunner (@embroider/webpack)
-~- created here: -~-
    at new Plugin (/Users/liam/Work/project/node_modules/.pnpm/broccoli-plugin@4.0.7/node_modules/broccoli-plugin/dist/index.js:47:36)
    at new PackagerRunner (/Users/liam/Work/project/node_modules/.pnpm/@embroider+core@3.4.3_@glint+template@1.2.2/node_modules/@embroider/core/src/to-broccoli-plugin.js:11:13)
    at Object.defaultPipeline (/Users/liam/Work/project/node_modules/.pnpm/@embroider+compat@3.4.3_@embroider+core@3.4.3_@glint+template@1.2.2/node_modules/@embroider/compat/src/default-pipeline.js:38:12)
    at module.exports (/Users/liam/Work/project/apps/admin/ember-cli-build.js:42:52)
    at Builder.readBuildFile (/Users/liam/Work/project/node_modules/.pnpm/ember-cli@5.5.0/node_modules/ember-cli/lib/models/builder.js:48:20)
    at async Builder.setupBroccoliBuilder (/Users/liam/Work/project/node_modules/.pnpm/ember-cli@5.5.0/node_modules/ember-cli/lib/models/builder.js:68:17)
    at async Builder.ensureBroccoliBuilder (/Users/liam/Work/project/node_modules/.pnpm/ember-cli@5.5.0/node_modules/ember-cli/lib/models/builder.js:56:7)
    at async Watcher.constructBroccoliWatcher (/Users/liam/Work/project/node_modules/.pnpm/ember-cli@5.5.0/node_modules/ember-cli/lib/models/watcher.js:59:5)
    at async Watcher.setupBroccoliWatcher (/Users/liam/Work/project/node_modules/.pnpm/ember-cli@5.5.0/node_modules/ember-cli/lib/models/watcher.js:43:37)
    at async Watcher.build (/Users/liam/Work/project/node_modules/.pnpm/ember-cli@5.5.0/node_modules/ember-cli/lib/models/watcher.js:31:5)
-~- (end) -~-
  - errorType: Build Error
  - location:
    - column: [undefined]
    - file: [undefined]
    - line: [undefined]
    - treeDir: [undefined]
  - message: ENOENT: no such file or directory, open '/var/folders/rw/zh799lvx7lzb46z1mh_r851m0000gn/T/broccoli-27890EVXZIM2sCKQ2/out-1227-packager_runner_embroider_webpack/admin/ember-cli-live-reload.js'
        at PackagerRunner (@embroider/webpack)
-~- created here: -~-
    at new Plugin (/Users/liam/Work/project/node_modules/.pnpm/broccoli-plugin@4.0.7/node_modules/broccoli-plugin/dist/index.js:47:36)
    at new PackagerRunner (/Users/liam/Work/project/node_modules/.pnpm/@embroider+core@3.4.3_@glint+template@1.2.2/node_modules/@embroider/core/src/to-broccoli-plugin.js:11:13)
    at Object.defaultPipeline (/Users/liam/Work/project/node_modules/.pnpm/@embroider+compat@3.4.3_@embroider+core@3.4.3_@glint+template@1.2.2/node_modules/@embroider/compat/src/default-pipeline.js:38:12)
    at module.exports (/Users/liam/Work/project/apps/admin/ember-cli-build.js:42:52)
    at Builder.readBuildFile (/Users/liam/Work/project/node_modules/.pnpm/ember-cli@5.5.0/node_modules/ember-cli/lib/models/builder.js:48:20)
    at async Builder.setupBroccoliBuilder (/Users/liam/Work/project/node_modules/.pnpm/ember-cli@5.5.0/node_modules/ember-cli/lib/models/builder.js:68:17)
    at async Builder.ensureBroccoliBuilder (/Users/liam/Work/project/node_modules/.pnpm/ember-cli@5.5.0/node_modules/ember-cli/lib/models/builder.js:56:7)
    at async Watcher.constructBroccoliWatcher (/Users/liam/Work/project/node_modules/.pnpm/ember-cli@5.5.0/node_modules/ember-cli/lib/models/watcher.js:59:5)
    at async Watcher.setupBroccoliWatcher (/Users/liam/Work/project/node_modules/.pnpm/ember-cli@5.5.0/node_modules/ember-cli/lib/models/watcher.js:43:37)
    at async Watcher.build (/Users/liam/Work/project/node_modules/.pnpm/ember-cli@5.5.0/node_modules/ember-cli/lib/models/watcher.js:31:5)
-~- (end) -~-
  - name: Error
  - nodeAnnotation: @embroider/webpack
  - nodeName: PackagerRunner
  - originalErrorMessage: ENOENT: no such file or directory, open '/var/folders/rw/zh799lvx7lzb46z1mh_r851m0000gn/T/broccoli-27890EVXZIM2sCKQ2/out-1227-packager_runner_embroider_webpack/admin/ember-cli-live-reload.js'
  - stack: Error: ENOENT: no such file or directory, open '/var/folders/rw/zh799lvx7lzb46z1mh_r851m0000gn/T/broccoli-27890EVXZIM2sCKQ2/out-1227-packager_runner_embroider_webpack/admin/ember-cli-live-reload.js'

=================================================================================

  ```
</details>

On these versions

```
"ember-source": "5.5.0",
"@embroider/compat": "3.4.3",
"@embroider/core": "3.4.3",
"@embroider/router": "2.1.6",
"@embroider/webpack": "3.2.1",
"webpack": "5.89.0",
```

**Comment by evoactivity** [comment]
I think it's not taking into account the rootUrl (admin in this case).

I tried on a different project and ran into runtime errors instead.

Firefox
```
The hash contained in the integrity attribute has the wrong length. [localhost:4200](http://localhost:4200/)
---
None of the “sha384” hashes in the integrity attribute match the content of the subresource. The computed hash is “unPTnL2TEPzMWsKAkmUpkuP2YDql+wGkog9XIV7ePFJYrpKcWHiZvCCybkQTM8LR”.
---
```

Chrome
```
Failed to find a valid digest in the 'integrity' attribute for resource 'http://localhost:4200/assets/project.css' with computed SHA-384 integrity 'unPTnL2TEPzMWsKAkmUpkuP2YDql+wGkog9XIV7ePFJYrpKcWHiZvCCybkQTM8LR'. The resource has been blocked
```

**Comment by evoactivity** [comment]
Changing the digest to `base64` instead of `hex` resolved my runtime errors.

**Comment by jelhan** [comment]
> I think it's not taking into account the rootUrl (admin in this case).

Good catch.

I think we have at least 2 errors:

1. The `ember-cli-livereload.js` should be skipped. But that skip does not work if `rootUrl` is set: https://github.com/jelhan/webpack-subresource-integrity-embroider/blob/36d2252a0f9934ed87eb128e85b2e6e2728e16ac/webpack-subresource-integrity-embroider/index.js#L27-L29
2. The working directory of Embroider is likely not containing the `rootUrl` causing the file lookup to fail. Could you please check if the `/assets/vendor.js` is in a subfolder or not?

Assuming that temporary working directory does _not_ include the `rootUrl`, we likely need to remove the `rootUrl` from the `href` / `src` value of the attribute. This should be done here: https://github.com/jelhan/webpack-subresource-integrity-embroider/blob/36d2252a0f9934ed87eb128e85b2e6e2728e16ac/webpack-subresource-integrity-embroider/index.js#L22-L25

> Changing the digest to `base64` instead of `hex` resolved my runtime errors.

Noticed that one myself in a reproduction app. Not covered by the test suite of the plugin. And likely introduced after I pulled out the changes from an app into this addon. Will fix it quickly.

For reference, it is about this line: https://github.com/jelhan/webpack-subresource-integrity-embroider/blob/36d2252a0f9934ed87eb128e85b2e6e2728e16ac/webpack-subresource-integrity-embroider/index.js#L32-L34

**Comment by evoactivity** [comment]
I have a fix incoming!

**Comment by jelhan** [comment]
> I have a fix incoming!

Adding support for applications having `rootUrl` set? :pray: 

Because I just pushed a fix for the invalid encoding of hash as `integrity` value is fixed in https://github.com/jelhan/webpack-subresource-integrity-embroider/pull/5. 😬

**Comment by jelhan** [comment]
Invalid encoding of has as `integrity` value should be fixed by [v0.1.2](https://github.com/jelhan/webpack-subresource-integrity-embroider/releases/tag/v0.1.2). Please confirm.

**Comment by evoactivity** [comment]
> Invalid encoding of has as `integrity` value should be fixed by [v0.1.2](https://github.com/jelhan/webpack-subresource-integrity-embroider/releases/tag/v0.1.2). Please confirm.

Confirmed. You did it exactly as I did it 😆
